TACC/Core-CMS
# feat: CMD-164 custom styles via setting

## Overview

New setting `PORTAL_STYLES` to load custom stylesheets.

## Related

- [CMD-164](https://tacc-main.atlassian.net/browse/CMD-164)
- documented by https://github.com/TACC/Core-CMS/pull/843
- required by https://github.com/TACC/Core-CMS-Custom/pull/292
- required by https://github.com/TACC/Core-CMS-Custom/pull/293
- required by https://github.com/TACC/Core-Portal-Deployments/pull/45

## Changes

- **added** setting `PORTAL_STYLES`
- **changed** template to load styles via `PORTAL_STYLES`

## Testing

1. Add `PORTAL_STYLES` to `taccsite_cms/settings_local.py` (or `…/camino/cms.settings_custom.py`):
    ```py
    PORTAL_STYLES = [{
        "is_remote": True,
        "path": "https://cdn.jsdelivr.net/gh/TACC/Core-CMS-Custom@2cdc59f/example_cms/src/apps/example_app/static/example_app/css/example_app.css",
    }]
    ```
2. Load CMS in browser.
3. Verify example stylesheet is loaded. 

## UI

| markup | network |
| - | - |
| <img width="1200" alt="markup" src="https://github.com/TACC/Core-CMS/assets/62723358/f559d8dc-27c5-4b06-ba68-9b17259e7f1b"> | <img width="1200" alt="network" src="https://github.com/TACC/Core-CMS/assets/62723358/9b99318f-ef1d-4134-8580-051d1829a194"> |

| remote |
| - |
| <img width="1200" alt="stylesheet" src="https://github.com/TACC/Core-CMS/assets/62723358/a82e4258-e9c8-498b-966a-9250074060ad"> |
